### ISSUE TITLE
commandline argument -oJunit disables -v

### ISSUE THREAD

**Issue by WilcoBT** [cpputest/cpputest#571]
I just noticed on our jenkins that the verbose output is not working. This seems to be caused by also having the option '-oJunit' active. It seems the inheritance of testOutput by JUnitTestOutput disables this functionality.

Is this accidental or is there a rationale behind this?

I'm already looking into fixing it myself.

**Comment by arstrube** [comment]
Hi there,

This behavior is intentional. One simple solution would be to run the tests twice, one time with -oJunit, the other time without.

As an aside, we found the JUnit option of little use. We are using the -v output for further parsing and report collection.

**Comment by WilcoBT** [comment]
the junit is useful for us to combine the results of different test projects run within one Jenkins job.
And running twice might work for now, but in the end we will need thousands of tests and I dislike the idea of then doubling the run time.

I do find it hard to verify my changes though, maybe because vs2010 is the one I do not have installed.

**Comment by ajf58** [comment]
We had the same problem. Our tests produce lots of output to the console, and having the test names written before each test is run really helps to find the problem. Like you @WilcoBT we use the xUnit output to give at-a-glance summaries of the project that integrated with our CI server.

In the end the solution I came up with was to write a simple plugin that did almost exactly what the [TestOutput](https://github.com/cpputest/cpputest/blob/master/src/CppUTest/TestOutput.cpp) class does and register it in our main function if both '-v' and '-oJunit' were defined. If you take a look at the [TestPlugin](https://github.com/cpputest/cpputest/blob/master/include/CppUTest/TestPlugin.h) class then you can see that all is needed is for the plugin's ```preTestAction``` and ```postTestAction``` methods to get the test case's name as is done [here](https://github.com/cpputest/cpputest/blob/master/src/CppUTest/TestOutput.cpp#L95). This method will ensure you don't have to run your tests twice, which would be very undesirable as you say.

Sorry for this verbose reply, but I'm not permitted to distribute the source code.

@arstrube is does seem like an odd design decision. The xUnit output in no way adversely affects what's written to the console, and it does seem like all the logging (default, verbose, xUnit) could be implemented as a stack of test plugins. If it really is a deliberate decision then it may be better to show the intent when parsing the command line arguments (e.g. throw an error like ```unable to support verbose flag when -ojunit is selected```) although that would be worse for me and @WilcoBT.

**Comment by arstrube** [comment]
@ajf58 I made it sound like it was _my_ design decision ;-) Actually, I have no idea who decided it or why, but my new PR #596 attempts to solve the problem in the simplest way I could think of. I gave the JUnitTestOutput a ConsoleTestOutput member and have it delegate console output to that, if -v is active. (I think there's not much point in having the dots printed out when doing -junit, is there?)

Seems like a very nifty idea to me to write a plugin to do the job :-).
